package.json

```diff
@@ -10,15 +10,29 @@
     "type": "git",
     "url": "https://github.com/MetaMask/metamask-module-template.git"
   },
-  "main": "./dist/index.js",
-  "types": "./dist/index.d.ts",
+  "exports": {
```

Comment:
Will this change how libraries can be exported going forward? Currently we are allowed to do `import { X } from "@metamask/y/dist/z"`.

Reply:
This is still allowed, but consumers have to pick whether they want the CJS or ESM version, i.e., `import { X } from '@metamask/y/dist/cjs/z'` or `'@metamask/y/dist/esm/z'`.

Reply:
@mcmire is correct, this does prevent referencing files inside the package. See https://nodejs.org/api/packages.html#package-entry-points

>The ["exports"](https://nodejs.org/api/packages.html#exports) provides a modern alternative to ["main"](https://nodejs.org/api/packages.html#main) allowing multiple entry points to be defined, conditional entry resolution support between environments, and **preventing any other entry points besides those defined in ["exports"](https://nodejs.org/api/packages.html#exports)**. This encapsulation allows module authors to clearly define the public interface for their package.

Reply:
Though I am OK with this being in our template. We try not to rely on referencing files directly in packages, since we tend not to consider that in determining which changes are breaking.

Reply:
Interesting. I was testing this in the extension, and it worked there, but that may be Browserify using their own `require` implementation. 🤔

Reply:
Adding an export for `./*` seems to resolve this.

Reply:
I'm okay with this being in our template as well, but maybe we reserve this for another PR just for visibility's sake.